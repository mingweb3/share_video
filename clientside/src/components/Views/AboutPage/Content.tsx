interface IContentProps {
  data: IPost
}

function Content({ data }: IContentProps) {
  return (
    <div className="pt-8">
      {data && (
        <article>
          {data.title && <h2 className="mb-4 text-2xl capitalize">{data.title}</h2>}
          {data.body && <div className="blog-ct text-md tracking-wider text-[#666666] lg:text-[24px]">{data.body}</div>}
        </article>
      )}
    </div>
  )
}

export default Content
