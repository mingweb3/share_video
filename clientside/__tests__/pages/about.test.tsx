/* eslint-disable import/no-extraneous-dependencies */
/**
 * @jest-environment jsdom
 */
import { render, screen } from '@testing-library/react'
import 'whatwg-fetch'
import { rest } from 'msw'
import { setupServer } from 'msw/node'

import AboutPage from '@/app/about/page'

const apiBase = process.env.API_BASE
const postMock = {
  userId: 1,
  id: 1,
  title: 'Title of the post ABC song',
  body: 'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto'
}
// Setup REQUEST SERVER
const handlers = [
  rest.get(`${apiBase}/posts/1`, (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(postMock))
  })
]
const server = setupServer(...handlers)

// Start request server
const consoleSpy = jest.spyOn(console, 'log').mockImplementation()
beforeAll(() => server.listen())
afterEach(() => {
  server.restoreHandlers()
  consoleSpy.mockClear()
})
afterAll(() => server.close())

// *** TESTING ***
describe('Render AboutPage', () => {
  it('First Heading', async () => {
    const jsx = await AboutPage()
    render(jsx)

    // Find Element to test
    const headingOne = screen.queryByRole('heading', { name: 'About us' })

    expect(headingOne).toBeInTheDocument()
    expect(headingOne).toHaveClass('mb-2 text-2xl font-bold')
  })

  it('Show Hero Banner', async () => {
    const jsx = await AboutPage()
    render(jsx)

    // Find Element to test
    const heroImg = screen.queryByRole('img')

    expect(heroImg).toBeInTheDocument()
    expect(heroImg).toHaveAttribute('src', '/images/hero.webp')
    expect(heroImg).toHaveAttribute('alt', 'about banner')
  })

  it('Render Content of Data', async () => {
    const jsx = await AboutPage()
    render(jsx)

    // Find Element to test: Title of article
    const header = screen.queryByRole('heading', { level: 2 })

    expect(header).toBeInTheDocument()
    expect(header).toHaveTextContent(postMock.title) // is the right content
    expect(header).toHaveClass('mb-4 text-2xl capitalize') // right style
  })

  it('ABOUT Page renders correctly with Snapshot', async () => {
    const jsx = await AboutPage()
    const { container } = render(jsx)
    expect(container).toMatchSnapshot()
  })
})
