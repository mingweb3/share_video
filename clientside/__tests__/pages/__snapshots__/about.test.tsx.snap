// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render AboutPage ABOUT Page renders correctly with Snapshot 1`] = `
<div>
  <div
    class="container mx-auto px-[20px] md:px-0"
  >
    <div
      class="inner pb-44 pt-6"
    >
      <h1
        class="mb-2 text-2xl font-bold"
      >
        About us
      </h1>
      <div
        class="picture h-[480px] overflow-hidden"
      >
        <img
          alt="about banner"
          src="/images/hero.webp"
        />
      </div>
      <div
        class="pt-8"
      >
        <article>
          <h2
            class="mb-4 text-2xl capitalize"
          >
            Title of the post ABC song
          </h2>
          <div
            class="blog-ct text-md tracking-wider text-[#666666] lg:text-[24px]"
          >
            quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
          </div>
        </article>
      </div>
    </div>
  </div>
</div>
`;
